tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/SparseTensor.java

```diff
@@ -0,0 +1,108 @@
+package org.tensorflow;
+
+import org.tensorflow.types.TInt64;
+import org.tensorflow.types.family.TType;
+
+/**
+ * A virtual type of {@link Tensor} composed of three dense tensors (indices, values and dimensions) used to represent
+ * the sparse data into a multi-dimensional dense space.
+ *
+ * Any tensor returned by a sparse tensor factory (e.g. {@link TInt64#sparseTensorOf(TInt64, TInt64, TInt64)}) can be casted back
+ * to this interface to access directly the dense tensors it is composed of.
+ *
+ * @param <T> type of data stored in the tensor
+ */
+public interface SparseTensor<T extends TType> extends Tensor {
```

Comment:
I had the same thought... but cannot find a good way to avoid. Ideally `Tensor` would also contain it's type but it is not possible since the type classes implement themselves the `Tensor` interface. :+1: for revisiting this later.